code/game/objects/structures/pinatas.dm

```diff
@@ -0,0 +1,83 @@
+/obj/structure/pinata
+	name = "corgi pinata"
+	desc = "A paper mache representation of a corgi that contains all sorts of sugary treats."
+	icon = 'icons/obj/toys/toy.dmi'
+	icon_state = "pinata_placed"
+	base_icon_state = "pinata_placed"
+	max_integrity = 300 //20 hits from a baseball bat
+	anchored = TRUE
+	///What sort of candy the pinata will contain
+	var/candy_options = list(/obj/item/food/candy, /obj/item/food/lollipop/cyborg, /obj/item/food/gumball, /obj/item/food/bubblegum, /obj/item/food/chocolatebar)
+	///How much candy is dropped when the pinata is destroyed
+	var/destruction_loot = 5
+	///Debris dropped when the pinata is destroyed
+	var/debris = /obj/effect/decal/cleanable/wrapping/pinata
+
+/obj/structure/pinata/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/pinata, candy = candy_options, death_drop = destruction_loot)
+
+/obj/structure/pinata/take_damage(damage_amount, damage_type, damage_flag, sound_effect, attack_dir, armour_penetration)
+	. = ..()
+	if(get_integrity() < (max_integrity/2))
+		icon_state = "[base_icon_state]_damaged"
+	if(damage_amount >= 10) // Swing means minimum damage threshhold for dropping candy is met.
+		flick("[icon_state]_swing", src)
+
+/obj/structure/pinata/play_attack_sound(damage_amount, damage_type, damage_flag)
+	switch(damage_type)
+		if(BRUTE)
+			if(damage_amount)
+				playsound(src, 'sound/weapons/slash.ogg', 50, TRUE)
+			else
+				playsound(src, 'sound/weapons/tap.ogg', 50, TRUE)
+		if(BURN)
+			playsound(src, 'sound/items/welder.ogg', 100, TRUE)
+
+/obj/structure/pinata/deconstruct(disassembled)
+	new debris(get_turf(src))
+	return ..()
+
+/obj/item/pinata
+	name = "pinata assembly kit"
+	desc = "a paper mache corgi that contains various candy, must be set up before you can smash it."
+	icon = 'icons/obj/toys/toy.dmi'
+	icon_state = "pinata"
+	///The pinata that is created when this is placed.
+	var/pinata_type = /obj/structure/pinata
+
+/obj/item/pinata/attack_self(mob/user)
+	var/turf/player_turf = get_turf(user)
+	if(player_turf?.is_blocked_turf(TRUE))
+		return FALSE
+	user.visible_message(span_info("[user] begins to set up \the [src]..."))
+	if(do_after(user, 4 SECONDS, target = get_turf(user), progress = TRUE))
+		new pinata_type(get_turf(user))
+		to_chat(user, span_notice("You set up \the [src]."))
+		qdel(src)
+
+/obj/structure/pinata/syndie
+	name = "syndicate corgi pinata"
+	desc = "A paper mache representation of a corgi that contains all sorts of bombastic treats."
+	icon_state = "pinata_syndie_placed"
+	base_icon_state = "pinata_syndie_placed"
+	destruction_loot = 2
+	debris = /obj/effect/decal/cleanable/wrapping/pinata/syndie
+	candy_options = list(
+		/obj/item/food/candy,
+		/obj/item/food/lollipop,
+		/obj/item/food/gumball,
+		/obj/item/food/bubblegum,
+		/obj/item/food/chocolatebar,
+		/obj/item/grenade/c4,
+		/obj/item/grenade/clusterbuster/soap,
+		/obj/item/grenade/empgrenade,
+		/obj/item/grenade/syndieminibomb,
+		/obj/item/grenade/frag
+		)
```

Comment:
```suggestion
		/obj/item/grenade/frag,
	)
```

also alphabetize this list

Reply:
Am I fine to keep the explosives separate from the candy item's when alphabetized? (A-Z candy followed by A-Z explosives) I feel like that might be easier to read in the long run?

Reply:
yeah sure just comment that so people know why you did it like that before they go "silly old coder didn't know what they were doing"